rmf_task/include/rmf_task/TaskPlanner.hpp

```diff
@@ -198,6 +198,23 @@ class TaskPlanner
     Configuration configuration,
     Options default_options);
 
+  /// Constructor
+  ///
+  /// \param[in] configuration
+  ///   The configuration for the planner
+  ///
+  /// \param[in] default_options
+  ///   Default options for the task planner to use when solving for assignments.
+  ///   These options can be overriden each time a plan is requested.
+  ///
+  /// \param[in] task_planner_name
+  ///   Identifier of this task planner, to be used for booking automated
+  ///   requests.
+  TaskPlanner(
+    Configuration configuration,
+    Options default_options,
+    const std::string& task_planner_name);
```

Comment:
It would be best to move the `task_planner_name` into the `Configuration` class and have a getter that returns `std::optional<std::string>`. The point of that class was to bundle all parameters that do not change between planning instances so it would be a good addition. Also consider calling the param `planner_id`? instead?

Reply:
`planner_id` is much better. It looks like `Configuration` stores parameters that are specifically used for planning, I think it would better to separate the concern and keep the identifier/name a part of the constructor of `TaskPlanner`. I would even bump the id to the first argument, similar to how we would name our `rclcpp::Node` first before giving it the node options.

Reply:
https://github.com/open-rmf/rmf_task/pull/81/commits/a1d26aa704d2ee224a6c41630a2a16af4b35b717